pkg/cmd/operator/webhooks.go

```diff
@@ -165,6 +175,13 @@ func (o *WebhookOptions) Run(streams genericclioptions.IOStreams, commandName st
 		cancel()
 	}()
 
+	return o.run(ctx, streams)
+}
+
+func (o *WebhookOptions) run(ctx context.Context, streams genericclioptions.IOStreams) error {
+	var wg sync.WaitGroup
+	defer wg.Wait()
```

Comment:
I would prefer explicit wg.Wait in code instead of in defer. Defer is good for cleanup, but not for logic

Reply:
how would you make sure it's called in all the branches? the point of defer here is to make sure we **always** wait for the children

Reply:
(it's the same as with locks, IMO)

Reply:
Now it hangs infinitely if L224 branch is triggered. IMO it's easier to spot this kind of deadlocks when blocking paths are mentioned explictly.

Reply:
We must track all our workers reliably and manual wait is a recipe for forgetting that. 
That said, we shouldn't hang - are you saying Shutdown won't cancel if `ListenAndServeTLS` fails?

We should add a cancelable context after the wg which would make sure no blocking call is left behind.

Reply:
guess that hangs on the `<-ctx.Done()` part because it's missing the cancalable context for this function, which was present before

Reply:
@rzetelskik this deserves a unit test - use an invalid port as a reproducer